sched/adaptdl_sched/policy/pollux.py

```diff
@@ -341,6 +386,10 @@ def _repair(self, pop, **kwargs):
         with np.errstate(divide="ignore", invalid="ignore"):
             states = np.amin(np.floor_divide(states, job_resources),
                              where=job_resources > 0, initial=99, axis=-1)
+        # Only choose solutions which have at least min_replicas allocations
+        min_replicas = np.array([j.min_replicas for j in self._jobs])
+        mask = np.sum(states, axis=-1) < min_replicas
+        states[mask] = 0
```

Comment:
I wonder if there is a head-of-line blocking issue with this approach.

For example, say there's a cluster with 16 GPUs. A non-preemptible job at position 0 wants 1 GPU. A non-preemptible job at position 1 wants 16 GPUs. And then followed by a bunch of preemptible jobs.

In this scenario, the repair step would first de-allocate all replicas from the preemptible jobs at positions 2, 3, ..., since the position 0 and 1 jobs are already assigned all the GPUs in the cluster. Next, the non-preemptible job at position 1 would have all its replicas de-allocated since it only has 15/16 of the replicas it wants. As a result, only 1 GPU in this cluster will be in use, until the position 0 job is finished and all 16 GPUs can go to the position 1 job.

Reply:
I supposed this should be solved by using a smarter allocation method than the `cumsum`/`diff` we do currently.

Reply:
non-preemptible jobs without allocations (not pinned) are treated exactly like preemptible jobs (FIFO ordering)

Reply:
the job sorting order, lower minReplicas ahead of higher minReplicas tries to avoid this.